### ISSUE TITLE
loginwindow never appears after loading bar with WEG 1.4.5. Dual RX580, 6 DP Displays

### ISSUE THREAD

**Issue by bu11etpr00f** [acidanthera/bugtracker#1407]
I have two XFX Radeon RX580 GTS BLK ED OC+ 1405MHz Graphics Cards with three DisplayPort Displays attached to each (six total displays).

The loginwindow WILL load when using WEG 1.4.4 or below (only tested 1.3.7 & 1.4.4).

The loginwindow WILL NOT load when using WEG 1.4.5.

I have attached both the debug log from when the loginwindow failed to load, and EFI (SN Removed) when using WEG 1.4.5 DEBUG version.

I am using an HP Z6 G4 with Dual Xeon Bronze 3104 CPU's and 64GB RAM configured as MacPro7,1.
macOS 10.15.7

Please let me know if there is anything else I can provide to help you resolve this issue.

EDIT: Updated macOS Version.
EDIT: Uploaded correct debug report for 1.4.5.

[EFI.no.SN.zip](https://github.com/acidanthera/bugtracker/files/5761308/EFI.no.SN.zip)
[weg.1.4.5.debug.log.txt](https://github.com/acidanthera/bugtracker/files/5761337/weg.1.4.5.debug.log.txt)

**Comment by vit9696** [comment]
Please tell us where it started (the commit). Use the builds from here: https://dortania.github.io/builds/?product=WhateverGreen&viewall=true. Make sure you have the latest DEBUG versions of all the other kexts. 

Once you find last working and first not working builds, attach the logs from them. Use `liludump=60 -liludbgall` to create a log in `/var/log/Lilu_x.x.x.txt` file 60 seconds after booting (increase 60 to whatever value that fits your hardware boot time).

**Comment by bu11etpr00f** [comment]
@vit9696

Will do. Is "Bump version" the first commit of a version, or is it the last?

Thanks!

**Comment by vit9696** [comment]
The newest version is top-most, thus `Bump version` is the first commit for each version (well, normally).

AppleMCEReporterDisabler should be a plist-only kext, thus no debug version for it.

**Comment by bu11etpr00f** [comment]
I really don't know where to start, and am a lot stuck. I think the issue has something to do with Catalina more than anything, because I can boot Mojave with no trouble no matter what version of WEG I use.

1.3.7 seems to work the best, but even it sometimes doesn't load the loginwindow. The strangest thing is when I get a strange overlay over the loginwindow. Even sometimes when only the loading bar seems stuck, I can hit several combinations of Tab, Esc, and Enter, and finally get it to load the loginwindow correctly, and then log in. Sometimes after logging in, the overlay is there and I can't get rid of it. If I ssh into the system and kill the WindowServer I can re-login, and everything will be fine...at least for a while. Sometimes, I just unplug one of the Displays, and plug it back in to correct the issue.

At first I thought it was related to a USB issue because it appeared that some modifier key was being stuck. e.g. the Edit menu would be highlighted in the Menu Bar in active apps. But after removing all USB drivers (RazerHID.kext, etc.), the issue still seemed to persist.

The other strange thing that was happening is that Application windows were not being updated. If I typed in Terminal, nothing would display until I either clicked on the window (which would sometimes work), or hit Enter several times after not seeing the expected result from a command.

I have tried the following without any resolve:
- Running First Aid on the Installation Drive.
- Updating to OC 0.6.5.
- Updating to the latest versions of the Lilu Plugins I'm using.
- Reverting back to the Lilu Plugins I used when I first started using OpenCore.
- Removing ACPI/SSDT-HPET.aml & SSDT-PMC.
- Removing all Kexts except AppleMCEReporterDisabler.kext, CpuTscSync.kext, Lilu.kext, SMCProcessor.kext, SMCSuperIO.kext, VirtualSMC.kext, & WhateverGreen.kext.
- Switching to MacPro6,1 (what my system was when I first started using OpenCore).
- (re)Installing 10.15.7 Combo Update.

It seems this issue first started after upgrading to the macOS Security Update after the 10.15.7 Update.

Suspecting that the issue might have something to do with corruption somewhere, I downloaded the 10.15.7 Combo Update and reinstalled it after adjusting the SystemVersion.plist for my installation back to 10.15.6.

IDK if something just got borked somewhere along the way, but I am pretty much at the end of my rope as far as troubleshooting goes. The only thing I can think of to try next is to Start with a fresh Catalina install. I'm just dreading having to do that. :(
But, it might be what I need to do...

I'm including all the DEBUG logs I have from the last few days, maybe you can find something that might be causing me this headache. [I am also including several photos of the weird overlay so you can see what I am talking about.](https://www.evernote.com/shard/s26/sh/1932181d-2087-11b7-5370-cbb82ac48be2/4d79dbd4f7eea2324cfe3bc5e3c0f946)

I labeled the Lilu DEBUG logs with each of the WEG versions I tried, and then with a number corresponding to the build from the link you sent me (1-15). 1 being the first, and 15 being the last.
[Lilu.DEBUG.logs.zip](https://github.com/acidanthera/bugtracker/files/5778656/Lilu.DEBUG.logs.zip)

I tried looking at the logs, but I don't see anything that makes any sense to me. Part of me wonders if maybe it's a framebuffer issue in Catalina, but IDK...

I'm also going to include all the Results for my system from SSDTime.
[SSDTTime.results.zip](https://github.com/acidanthera/bugtracker/files/5778681/SSDTTime.results.zip)

Thanks!

**Comment by vit9696** [comment]
I get a feeling there is some race condition in Apple drivers with your system. The overlay you see is apparently caused by AppleGraphicsDisplayPolicy, so trying different ones may have different effects. Also, are you sure all the GPU paths exist in ACPI for your system? If they do not, it is a mistake.

**Comment by bu11etpr00f** [comment]
> I get a feeling there is some race condition in Apple drivers with your system. The overlay you see is apparently caused by AppleGraphicsDisplayPolicy, so trying different ones may have different effects. Also, are you sure all the GPU paths exist in ACPI for your system? If they do not, it is a mistake.

I'm not sure how to try different race conditions, is there a guide you could point me to that would show me how to check what is being used, and how to set them? AFAIK the GPU paths exist, like I said, the same OC config works to boot Mojave without issue. What would I look for to confirm that the paths exist? Would I look for that in IORegExplorer?

Did you see anything glaringly wrong in any of the logs I attached, or any possible issues with the DSDT / SSDT Outputs from SSDTime?

I know this isn't a support forum, and I now know this may not even be a bug... I just have no way of knowing for sure since it's such a strange issue, and one of which I have never encountered in the 14 years I've been hackint0shing. I'm usually really good at troubleshooting issues, especially if it's a kext or hardware issue, this one has thrown me for a loop though. When I first opened this issue, I was sure it was something to do with WEG, but I just really don't know enough about ACPI, DSDT, and Bug Reports to confirm anything.

So any help you are willing to continue to offer is greatly appreciated!

Thanks!

**Comment by vit9696** [comment]
> I'm not sure how to try different race conditions, is there a guide you could point me to that would show me how to check what is being used, and how to set them? AFAIK the GPU paths exist, like I said, the same OC config works to boot Mojave without issue. What would I look for to confirm that the paths exist? Would I look for that in IORegExplorer?

To ensure that all GPUs are property described in ACPI use IORegistryExplorer and check that all GPUs (e.g. GFX0) have acpi-path property. If any does not, this must be fixed. You can override AGDP policy via agdpmod boot argument, see WEG README for possible values.

The logs look weird. `Lilu_1.5.0_19.6-145-15b-darkened.display.txt` does have IOGraphicsFamily patches, while `Lilu_1.5.0_19.6-145.final.txt` does not. But normally they all should. So there clearly is some initialisation issue with your system, likely with ACPI. I can suggest using `gfxrst=4` boot argument to disable some of the workarounds that might conflict with your configuration, but I am unsure if it is a good idea.

**Comment by bu11etpr00f** [comment]
I just wanted to give you an update on several of the things I have done to try to eliminate any possible issue on my side. ( I have not tried the gfxrst=4 boot argument yet, but I will in my next go around!)

1. I completely reinstalled Catalina on a new drive. No Resolve. The same issues are present on the new install, so it doesn't look like an issue with my existing install. From here I started looking at what might have changed since having a stable Catalina installation. The only change I could remember for sure was the Security Update that was released after 10.15.7, but with the new install that has been ruled out.
2. I tried switching back to the "Prebuilt" SSDTs (SSDT-PLUG & SSDT-EC-USBX) I had initially before creating my own with SSDTime. No Resolve.
3. I tried switching back to all the kext versions I was using when I remember having a stable installation. No Resolve.
4. I manually built new SSDTs for my system, which I will upload for you to take a look at (I will also include the DSL files so you don't have to convert the AMLs). No Resolve with this either.
It's odd because some of the SSDT files I created were files the Skylake Guide said I didn't need, but the DSDT issues they addressed had the components needing patching present in my DSDT.

5. I tried changing obscure BIOS / Firmware settings. No Resolve, and the changes also didn't seem to change the DSDT I was pulling from my system. The settings I tried messing with (which for some I really have no idea what they should be set to (there isn't much help that I could find in the manual, or from a Search)) are:

- PCIe Training Reset: Disable (default) / Enable
- PCIe ACS: Enable (default) / Disable
- Runtime Power Management: Enable (default) / Disable
`Processor P States`
- Hardware P-States: Enable (default) / Disable
`Disable: P-State based on OS request (Legacy) / Enable: based on OS guidance.`
- Energy / Performance Bias Control: OS Controls EPB (default) / BIOS Controls EPB
- Idle Power Savings: Extended (default) / Normal / Normal with Enhanced Halt State disabled
- Non-Uniform Memory Access (NUMA): Enable (default) / Disable
- Sub-NUMA Clustering: Disable (default) / Enable
- isoc Mode: Disable (default) / Enable
`Disable: Higher Memory performance for Uniprocessor configurations.`

Currently I have all the above settings set to the defaults. If anything sticks out to you as something that could possibly be related to my issue, please let me know.

The next things I am going to try (in addition to your boot argument), are:
1. Roll back to OpenCore 0.6.4 and 0.6.3 (as well as the kexts I was using at that time).
2. There was something else, but I can't remember ATM.

The crazy things are:
1. Mojave works (I have it installed on a separate removable SSD Drive. I have a (4x2.5) 3.5 Removable Drive Bay ).
2. When the display doesn't load / has an overlay / is stuck on the Apple loading bar, hitting (MANY) random combinations of Tab, random letter Keys, Esc, random letter Keys, Enter, etc, etc, etc, repeat. The loginwindow will eventually load. If after logging in there is another hidden overlay, disconnecting one of the DP Cables, and reconnecting it seems to get rid of the overlay.

Part of me suspected that somehow the issue might be related to USB because of crazy issue #2, but after everything I learned while manually making my SSDTs from my DSDT, I'm not sure how it could be (at least it's over my understanding).

The files I previously uploaded (SSDTTime.results.zip & lspci.output.txt) above were created while booted to Ubuntu, just so you know that I didn't create while booted to macOS, so they should be unaltered by OpenCore from the BIOS / Firmware in any way.

IDK if you can tell If there is anything strange in regard to my DSDT that I might be overlooking or not.

Also, could you tell me what to look for in the logs when you said (just so I know what to look for on my own):
`The logs look weird. Lilu_1.5.0_19.6-145-15b-darkened.display.txt does have IOGraphicsFamily patches, while Lilu_1.5.0_19.6-145.final.txt does not.`

I'm not sure when the IOGraphicsFamily patch loads and is logged, but one thing I will mention is that with some of the logs I had `liludump=120` before I realized I needed to set it to `liludump=180` to get everything from the start of OC up to the loginwindow. So maybe that could be why it's missing???

[ACPI.zip](https://github.com/acidanthera/bugtracker/files/5839141/ACPI.zip)

**Comment by vit9696** [comment]
The time in liludump might have been it.

Regarding your ACPI, do you have both GPUs have a dedicated device in I/O Registry? I.e. each GPU (i.e. GFX0) should have acpi-path property in it. If it does not, it must get an SSDT. Check OpenCore DeviceProperties section in the pdf for more details.

**Comment by bu11etpr00f** [comment]
> The time in liludump might have been it.
> 
> Regarding your ACPI, do you have both GPUs have a dedicated device in I/O Registry? I.e. each GPU (i.e. GFX0) should have acpi-path property in it. If it does not, it must get an SSDT. Check OpenCore DeviceProperties section in the pdf for more details.

I believe they are. Here is a copy of the PCI Devices exported via Hackintool.

One is `PciRoot(0x3)/Pci(0x0,0x0)/Pci(0x0,0x0)` or `_SB_.PC02.BR2A.GFX0`
and the other is `PciRoot(0x2)/Pci(0x0,0x0)/Pci(0x0,0x0)` or `_SB_.PC03.BR3A.GFX1`

(Depending on whether you are looking at a txt or dsl file. I have also included a json and plist output as well.)

[pcidevices.output.hackintool.zip](https://github.com/acidanthera/bugtracker/files/5840712/pcidevices.output.hackintool.zip)

**Comment by bu11etpr00f** [comment]
Just a quick note. I loaded up a backup of the original OC 0.6.3 config I know I was running when everything was stable, and everything is stable again. I will run this config for a few days to confirm it stays stable, then I will start swapping stuff till it breaks again. That way I can tell you exactly where the fault is. I'm uploading a copy of my functioning EFI in the case that you want to take a gander.

Differences:
- OC 0.6.3 WORKS

- ACPI
     - SSDT-AWAC (disabled in config)
     - SSDT-EC-USBX-DESKTOP
     - SSDT-PLUG-DRTNIA

- Kexts
     - AppleALC 1.5.4
     - HibernationFixup 1.3.7
     - Lilu 1.4.9
     - SMCProcessor 1.1.8
     - SMCSuperIO 1.1.8
     - VirtualSMC 1.1.8
     - WhateverGreen 1.4.4

- Config
     - XhciPortLimit: false (doesn't seem to make a difference either way.)
     - BootProtect: Bootstrap



- OC 0.6.4 / 0.6.5 BROKEN

- ACPI
     - Everything I've changed here is listed throughout this entire thread, so I'm not going to list it all again here. :)

- Kexts
     - AppleALC 1.5.5
     - HibernationFixup (I've tried this config with and without this kext, appears to be irrelevant.)
     - Lilu 1.5.0
     - SMCProcessor 1.1.9
     - SMCSuperIO 1.1.9
     - VirtualSMC 1.1.9
     - WhateverGreen 1.4.5

- Config
     - XhciPortLimit: true (doesn't seem to make a difference either way.)
     - BootProtect: None


The only other differences in the Config are the things changed / added / removed between the config.plist versions themselves.


Here are the Kexts that are the same:
- Kexts
     - AppleMCEReporterDisabler
     - CpuTscSync 1.0.3
     - NoTouchID 1.0.4

[EFI.sn.removed.zip](https://github.com/acidanthera/bugtracker/files/5841479/EFI.sn.removed.zip)

**Comment by bu11etpr00f** [comment]
Now I'm almost positive I've isolated the culprit: CpuTscSync.

I loaded up my old 0.6.3 config and everything was back to normal. While testing changes to that config, and comparing it to the backup, I realized CpuTscSync was set to disabled in the working 0.6.3 config; but because CpuTscSync was in the Kexts folder, ProperTree (re)enabled it when doing a snapshot. As soon as I discovered this, I deleted CpuTscSync, updated EVERYTHING to the most current versions and booted. So far so good. I'll keep an eye on it for a few days just to be sure, but I'm pretty sure that was it. Wholly crap what a PITA!

Let me know if there is anything you might need for CpuTscSync. If everything runs smoothly for the next couple days, then I will be confident this was the issue. I'm not sure if that would classify as a bug, or just incompatibility. Either way, I'm sure this thread will help another tortured soul.
